### ISSUE TITLE
Species `OnEdited` should ensure `IsBacteria` is always up to date

### ISSUE THREAD

**Issue by hhyyrylainen** [Revolutionary-Games/Thrive#5304]
to ensure there's no editing or mutation bugs that don't correctly update the bacteria-ness of a species

**Comment by EuroBlue** [comment]
What do you mean with up to date?

**Comment by hhyyrylainen** [comment]
The species class does not update `public bool IsBacteria { get; set; }` so after an update of the organelles there is a potential bug if the updating code forgot to update the `IsBacteria` property.

It would be better if the species object took care of making sure there can not be a situation where the organelles are updated but the `IsBacteria` property is not and has an old value.

**Comment by IdfbAn** [comment]
What determines whether or not a species is a bacteria? Is every species without a nucleus a bacteria, or are there more conditions?

**Comment by hhyyrylainen** [comment]
Yes, anything with a nucleus is considered an eukaryote. There's no other conditions to check.

**Comment by IdfbAn** [comment]
So prokaryotes and bacteria are one and the same? (In game)

**Comment by hhyyrylainen** [comment]
Yes, the words are used as synonyms in the game.
